packages/edit-site/src/components/global-styles/typography-utils.js

```diff
@@ -172,6 +174,11 @@ export function getTypographyValueAndUnit( rawValue, options = {} ) {
 		return null;
 	}
 
+	// Converts numbers to pixel values by default.
+	if ( typeof rawValue === 'number' && ! Number.isNaN( rawValue ) ) {
```

Comment:
Checking for `NaN` is probably overkill. Unless it's a domain specific math function I don't think it's necessary to check for `Infinity`, `-Infinity`, `NaN` in functions that accept a number.